### DISCUSSION TITLE
Read chunked files from URLs

### DISCUSSION THREAD

**Discussion by dmkl** [ananthakumaran/zstream#17] [Ideas]
Hi @ananthakumaran,

we need to stream large files from a cloud storage. Is there an easy way to adopt `zstream` for feeding it read chunks? I'm not that advanced with Elixir yet. Can you give some hints on how to extend `zstream` for that purpose, please? Would it fit here at all?

**Comment by ananthakumaran** [comment]
I would likely need more info, whether zip/unzip, which cloud provider etc. But in most cases, you need to create the input/output as a stream. For example, if you want to download and zip large files from [s3](https://github.com/aws-beam/aws-elixir/issues/58), you can use the aws library to fetch the files and then use zstream to zip. In general, I would suggest getting familiar with Elixir stream abstraction, then the solution is usually a matter of composing the streams correctly.

**Reply by dmkl** [reply]
Thank you! I've figured out how to feed `Stream.resource` as a `zstream` entry. It works well 👍🏻
